src/metrics/family.rs

```diff
@@ -207,6 +207,40 @@ impl<S: Clone + std::hash::Hash + Eq, M, C> Family<S, M, C> {
     }
 }
 
+impl<S: Clone + std::hash::Hash + Eq, M: Clone, C: MetricConstructor<M>> Family<S, M, C>
+where
+    S: Clone + std::hash::Hash + Eq,
+    M: Clone,
+    C: MetricConstructor<M>,
+{
+    /// Access a metric with the given label set, creating it if one does not yet exist.
+    ///
+    /// ```
+    /// # use prometheus_client::metrics::counter::{Atomic, Counter};
+    /// # use prometheus_client::metrics::family::Family;
+    /// #
+    /// let family = Family::<Vec<(String, String)>, Counter>::default();
+    ///
+    /// // Will create and return the metric with label `method="GET"` when first called.
+    /// family.get_or_create_clone(&vec![("method".to_owned(), "GET".to_owned())]).inc();
+    ///
+    /// // Will return a clone of the existing metric on all subsequent calls.
+    /// family.get_or_create_clone(&vec![("method".to_owned(), "GET".to_owned())]).inc();
+    /// ```
+    ///
+    /// Callers wishing to avoid a clone of the metric `M` can call [`Family::get_or_create()`] to
+    /// return a reference to the metric instead.
+    pub fn get_or_create_clone(&self, label_set: &S) -> M {
```

Comment:
```suggestion
    pub fn get_or_create_owned(&self, label_set: &S) -> M {
```

Would this be more intuitive? I am undecided.

Reply:
i'm not strongly opinionated about the name, i went with `get_or_create_clone()` per your [previous suggestion](https://github.com/prometheus/client_rust/issues/231#issuecomment-2440130338). i'm happy with either. :slightly_smiling_face: :+1:

my ideal, personally, would be the interface described [here](https://github.com/prometheus/client_rust/issues/231#issuecomment-2415001551), because exposing read and write guards is prone to deadlocks, but that would be a breaking change at this point.

Reply:
Given that we call `to_owned`, I think the rename suggest above is more accurate. Thus I am in favor of `get_or_create_owned`. Sorry for the trouble. Mind doing the rename?